CesiumUtility/test/TestTreeTraversalState.cpp

```diff
@@ -0,0 +1,481 @@
+#include <CesiumUtility/TreeTraversalState.h>
+
+#include <doctest/doctest.h>
+
+#include <string>
+
+using namespace CesiumUtility;
+
+namespace {
+
+struct Node {
+  std::string name;
+};
+
+} // namespace
+
+TEST_CASE("TreeTraversalState") {
+  SUBCASE("Three Levels") {
+    Node a{"a"}, b{"b"}, c{"c"}, d{"d"}, e{"e"}, f{"f"};
+    TreeTraversalState<Node*, int> traversalState;
+
+    /*
+     *       a
+     *     / | \
+     *    b  c  d
+     *      / \
+     *     e   f
+     */
+    // clang-format off
+    traversalState.beginNode(&a);
```

Comment:
Definitely a nitpick but perhaps nesting scopes with `{}` could help with the readability / formatting?

Reply:
This is a good idea, but I tried it out on one of the blocks, and I think I prefer it as it is. The braces are nice because clang-format knows how to handle format it, but they add a like of vertical space and I don't think make it particularly easier to read. Or maybe I'm just lazy and don't feel like doing it everywhere. 😁